src/daemon/messages.go

```diff
@@ -280,6 +280,20 @@ func (intro *IntroductionMessage) Handle(mc *gnet.MessageContext, daemon interfa
 		}
 
 		if port == intro.Port {
+			if d.Pool.Pool.IsDefaultConnection(mc.Addr) {
```

Comment:
This will prevent default connections from exceeding 1 even if the default peer is incoming.  This is ok but the config value is named `MaxDefaultOutgoingConnections` so maybe change it to `MaxDefaultPeerConnections`

Reply:
No, it's actually only the outgoing connection, the `if port == intro.Port` here filters out incoming connections, as the incoming connection's port is a random port, it can't be the same as the port in introduction message.